### PR TITLE
feat(schema, vite)!: enable `vite-node` by default

### PR DESCRIPTION
**Pull request by antfu**
### 🔗 Linked issue

#5018

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/vite/src/vite.ts [hunk 1]
```diff
@@ -16,6 +16,7 @@ import { composableKeysPlugin } from './plugins/composable-keys'
 export interface ViteOptions extends InlineConfig {
   vue?: Options
   ssr?: SSROptions
+  viteNode?: boolean
```
Comment (pi0) [inline packages/vite/src/vite.ts:19]: Can we please change option to `serverBundler: 'vite-node' | 'legacy'`? This reflects the fact it is only for server and allows any new mechanism in the future as same option.

File: packages/schema/src/config/experimental.ts [hunk 2]
```diff
@@ -30,5 +25,20 @@ export default {
...
+
+    /**
+     * Use vite-node for on-demand server chunk loading
+     * 
+     * @deprecated use `vite.viteNode`
+     */
+    viteNode: {
+      $resolve: (val) => {
+        if (val === true)
+          console.warn('`vite-node` is now enabled by default. You can safely remove `experimental.viteNode` from your config.')
```
Comment (pi0) [inline packages/schema/src/config/experimental.ts:37]: This is nice with experimental notice. Can we also support `EXPERIMENTAL_VITE_NODE` and also code style: `if (...) {} else {}`

**Comment by misaon** [comment]
This will broke all guys which use docker for develop in Nuxt:
https://github.com/nuxt/framework/issues/3998#issuecomment-1184281675

**Comment by misaon** [comment]
@pi0 I can create a sandbox where the bug can be simulated. However, as has already been written in the thread, the bug manifests itself randomly. It doesn't seem to have an obvious cause, it's just that during development (editing components, pages, composables) a situation arises where the bug appears. In our team, where we develop in Docker projects built on Nuxt, the problem happens randomly to everyone.

We have found that if this error appears in the console, we have to make a change in some vue file (sometimes several times) to get the watcher to start again and work properly. We use the simplest form of Dockerfile

```dockerfile
FROM node:18.6.0-alpine3.15 as nuxtBuild

WORKDIR /app

COPY ./ ./

RUN yarn install \
    && yarn run build

FROM node:18.6.0-alpine3.15

ENV NODE_OPTIONS='--max-http-header-size=32768'
ENV HOST=0.0.0.0
WORKDIR /app

COPY --from=nuxtBuild /app/.output /app/.output
```

It is possible that this is the fault of the Alpine distribution itself, but it is on this distribution that NodeJS officially publishes its docker images.

**Comment by pi0** [comment]
I can confirm the same issue. It seems not even limited to docker but any Linux/Windows env, HMR watcher is ignored for server and only applied to the client, therefore, having hydration mismatch.

I've converted PR to draft until we solve this. 

It might be reload issue #2047 since now client HMR on linux works and Server not (in current nuxt, it is otherwise). But also trying `CHOKIDAR_USEPOLLING=1` I still have issue.

**Comment by misaon** [comment]
> 

Good work @pi0! ❤️

File: packages/schema/src/types/config.ts [hunk 3]
```diff
@@ -45,4 +45,9 @@ export interface ViteConfig extends ViteUserConfig {
    * @see https://github.com/vitejs/vite/tree/main/packages/plugin-vue
    */
   vue?: VuePluginOptions
+  /**
+   * Bundler for dev time server-side rendering.
+   * @default 'vite-node'
+   */
+  serverBundler?: 'vite-node' | 'legacy'
```
Comment (danielroe) [inline packages/schema/src/types/config.ts:52]: Maybe:


```suggestion
  devBundler?: 'vite-node' | 'legacy'
```

To make it clear we are not talking about anything runtime in built app.

**Review by pi0 — APPROVED** [review]
LGTM at current status. I cannot reproduce (at least stabily) HMR issues. I think we can move it forward to test on edge until next rc.

**Pull request merged**